core/src/main/java/org/apache/iceberg/io/datafile/ReadBuilder.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.io.datafile;
+
+import java.nio.ByteBuffer;
+import java.util.Map;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.expressions.Expression;
+import org.apache.iceberg.io.CloseableIterable;
+import org.apache.iceberg.mapping.NameMapping;
+
+/**
+ * Builder API for reading Iceberg data files.
+ *
+ * @param <R> type of the reader
+ * @param <F> type of the records which are filtered by {@link DeleteFilter}
+ */
+public interface ReadBuilder<R extends ReadBuilder<R, F>, F> {
+  /**
+   * Restricts the read to the given range: [start, start + length).
+   *
+   * @param newStart the start position for this read
+   * @param newLength the length of the range this read should scan
+   */
+  R split(long newStart, long newLength);
+
+  /** Read only the given columns. */
+  R project(Schema newSchema);
+
+  /**
+   * Sets the reader to case-sensitive when matching column names. Readers might decide not to
+   * implement this feature. The default is behavior is case-sensitive.
+   */
+  default R caseInsensitive() {
+    return caseSensitive(false);
+  }
+
+  default R caseSensitive(boolean newCaseSensitive) {
+    // Just ignore case sensitivity if not available
```

Comment:
I don't think this can simply ignore options.

Reply:
Avro ignores this currently. This is only filtering related. We might be able to rename the method to reflect this.